dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/ProgramTrackedEntityAttributeStore.java

```diff
@@ -39,4 +42,6 @@
     extends IdentifiableObjectStore<ProgramTrackedEntityAttribute>
 {
     ProgramTrackedEntityAttribute get( Program program, TrackedEntityAttribute attribute );
+
+    List<TrackedEntityAttribute> getAttributesByProgramsWithDataRead( List<Program> programs, User user );
```

Comment:
1) Please add Javadoc
2) I would rename this method to `getAttributes` and explain the ACL part in the java docs

Reply:
Fixed and added unit test